flink-runtime/src/test/java/org/apache/flink/runtime/webmonitor/threadinfo/ThreadInfoRequestCoordinatorTest.java

```diff
@@ -265,15 +267,20 @@ private static CompletableFuture<TaskExecutorThreadInfoGateway> createMockTaskMa
         return CompletableFuture.completedFuture(executorGateway);
     }
 
-    private static Map<Set<ExecutionAttemptID>, CompletableFuture<TaskExecutorThreadInfoGateway>>
+    private static Map<
+                    ImmutableSet<ExecutionAttemptID>,
+                    CompletableFuture<TaskExecutorThreadInfoGateway>>
             createMockSubtaskWithGateways(CompletionType... completionTypes) {
-        final Map<Set<ExecutionAttemptID>, CompletableFuture<TaskExecutorThreadInfoGateway>>
+        final Map<
+                        ImmutableSet<ExecutionAttemptID>,
+                        CompletableFuture<TaskExecutorThreadInfoGateway>>
                 result = new HashMap<>();
         for (CompletionType completionType : completionTypes) {
             Set<ExecutionAttemptID> attemptIds = new HashSet<>();
```

Comment:
You should be able to avoid `attemptIds` entirely. Can't you use `ImmutableSet.of(new ExecutionAttemptID(), new ExecutionAttemptID())`?